assets/js/modules/analytics-4/datastore/custom-dimensions-gathering-data.js

```diff
@@ -303,6 +306,90 @@ const baseSelectors = {
 			return false;
 		}
 	),
+
+	/**
+	 * Gets the options for the data availability report.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state           Data store's state.
+	 * @param {string} customDimension Custom dimension slug.
+	 * @return {Object|undefined} Returns the report options if the property is loaded, otherwise undefined.
+	 */
+	getDataAvailabilityReportOptions: createRegistrySelector(
+		( select ) => ( state, customDimension ) => {
+			invariant( customDimension, 'customDimension is required.' );
+
+			const propertyID = select( MODULES_ANALYTICS_4 ).getPropertyID();
+
+			if ( ! propertyID ) {
+				return undefined;
+			}
+
+			const property =
+				select( MODULES_ANALYTICS_4 ).getProperty( propertyID );
+
+			if ( property === undefined ) {
+				return undefined;
+			}
+
+			const startDate = getDateString( new Date( property.createTime ) );
+
+			const endDate = select( CORE_USER ).getReferenceDate();
+
+			return {
+				startDate,
+				endDate,
+				dimensions: [ `customEvent:${ customDimension }` ],
+				metrics: [ { name: 'eventCount' } ],
+				limit: 2,
+			};
+		}
+	),
+
+	/**
+	 * Gets the errors for the data availability report.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state            Data store's state.
+	 * @param {Array}  customDimensions Array of custom dimension slugs.
+	 * @return {Object} Returns an object of errors keyed by custom dimension slug.
+	 */
+	getDataAvailabilityReportErrors: createRegistrySelector(
+		( select ) => ( state, customDimensions ) => {
+			invariant( customDimensions, 'customDimensions is required.' );
+			invariant(
+				Array.isArray( customDimensions ),
+				'customDimensions must be an array.'
+			);
+
+			return customDimensions.reduce( ( errors, customDimension ) => {
+				const reportArgs =
+					select(
+						MODULES_ANALYTICS_4
+					).getDataAvailabilityReportOptions( customDimension );
+
+				if ( ! reportArgs ) {
+					return errors;
+				}
+
+				const error = select( MODULES_ANALYTICS_4 ).getErrorForSelector(
```

Comment:
Please get the selectors outside of the per-dimension loop rather than part of every iteration (i.e. outside of `.reduce`).

Reply:
Updated, thank you!